### ISSUE TITLE
[Bug]: iOS browser crashed when loading some pdfs

### ISSUE THREAD

**Issue by KKzLEO** [mozilla/pdf.js#18985]
### Attach (recommended) or Link to PDF file

[test.pdf](https://github.com/user-attachments/files/17589056/test.pdf)

### Web browser and its version

chrome 130.0.6723.78

### Operating system and its version

iOS 18.0.1

### PDF.js version

v4.7.76

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1. Go to `https://mozilla.github.io/pdf.js/web/viewer.html`
2. open `test.pdf` I provided
3. After loading a few pages, the entire browser crashed

https://github.com/user-attachments/assets/765f2e4a-be7f-4f7e-8c5a-a734d014d7d8
https://github.com/user-attachments/assets/4a9d6166-2ff3-4ffc-ad1d-c922a556e5ba

### What is the expected behavior?

Do not crash

### What went wrong?

crash

**Comment by KKzLEO** [comment]
Thank you for your response and the hard work your team is putting in. I am curious about the cause of this issue, especially since other PDFs with larger image sizes are functioning correctly. I also tried loading the pages one by one by removing the previous pages and adding new ones. Initially, this approach worked well, but it crashed after a few pages. Generally, I believe other pages should be garbage collected, correct? However, this doesn't seem to work effectively in Safari. Could there be memory leak issues with PDF.js? Other PDF frameworks, such as PSPDFKit, seem to perform well.

**Comment by KKzLEO** [comment]
Hello @Snuffleupagus , I hope you're doing well. It seems that this issue might not be exclusive to Safari, as other PDF.js-based libraries, such as PDF.js Express, are working correctly. Could you kindly consider reopening this issue? Thank you for your attention.

**Comment by marco-c** [comment]
@KKzLEO PDF.js Express is using a very old version of PDF.js, if it is working there it means this might be a regression from one of our changes. Could you try with various versions of PDF.js to isolate which one introduced this problem?

**Comment by KKzLEO** [comment]
@marco-c 

<img width="1301" alt="image" src="https://github.com/user-attachments/assets/53a4dc86-776b-431d-99e7-fd50d6b2ffd5" />
Hi, I found that there's a weird `OffscreenCanvas` didn't got cleared when we opened the pdf, it might be the reason why iOS always crashed. The steps to reproduce the problem is the same but we have to connect iOS device with mac by wire and open safari in mac to debug.

Steps:

1. Connect iOS device with mac by wire
2. Go to https://mozilla.github.io/pdf.js/web/viewer.html
3. open test.pdf I provided
4. open safari in mac and enter debugging mode
5. check the graphic tab
6. after scrolling, you might see the weird `OffScreen2D`, which is very big, I think it's the root cause
